File: src/HelloTizenFormFactory.cpp
#include "HelloTizenFormFactory.h"
#include "HelloTizenMainForm.h"
#include "AppResourceId.h"

using namespace Tizen::Ui::Scenes;


HelloTizenFormFactory::HelloTizenFormFactory(void)
{
}

HelloTizenFormFactory::~HelloTizenFormFactory(void)
{
}

Tizen::Ui::Controls::Form*
HelloTizenFormFactory::CreateFormN(const Tizen::Base::String& formId, const Tizen::Ui::Scenes::SceneId& sceneId)
{
	SceneManager* pSceneManager = SceneManager::GetInstance();
	AppAssert(pSceneManager);
	Tizen::Ui::Controls::Form* pNewForm = null;

	if (formId == IDF_FORM)
	{
		HelloTizenMainForm* pForm = new HelloTizenMainForm();
		pForm->Initialize();
		pSceneManager->AddSceneEventListener(sceneId, *pForm);
		pNewForm = pForm;
	}
	// TODO:
	// Add your form creation code here

	return pNewForm;
}

File: HelloTizen/Source/HelloTizenFormFactory.cs

using Tizen;
using Tizen.UI.Scenes;


public class HelloTizenFormFactory : Tizen.UI.Scenes.IFormFactory
{
	public Tizen.UI.Controls.Form CreateFormN (Tizen.Base.String formId, Tizen.UI.Scenes.SceneId sceneId)
	{
		SceneManager sceneManager = SceneManager.GetInstance();
		Utilities.AppAssert(sceneManager != null);
		Tizen.UI.Controls.Form newForm = null;

		if (formId.Equals(Constants.IDF_FORM))
		{
			//HelloTizenMainForm form = new HelloTizenMainForm();
			//form.Initialize();
			//sceneManager.AddSceneEventListener(sceneId, form);
			//newForm = form;
		}
		// TODO:
		// Add your form creation code here

		return newForm;
	}
}
